### ISSUE TITLE
[Bug] [Server] delayed execution is useless

### ISSUE THREAD

**Issue by DHBin** [apache/dolphinscheduler#8480]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

设置 延时执行时间 为 2，运行任务，任务立刻执行了

### What you expected to happen

设置 延时执行时间 为 2，运行任务，任务等待2分钟才执行

### How to reproduce

设置 延时执行时间 为 2，运行任务

### Anything else

```java
    public TaskInstance submitTaskInstanceToDB(TaskInstance taskInstance, ProcessInstance processInstance) {
        ExecutionStatus processInstanceState = processInstance.getState();

        if (taskInstance.getState().typeIsFailure()) {
            if (taskInstance.isSubProcess()) {
                taskInstance.setRetryTimes(taskInstance.getRetryTimes() + 1);
            } else {
                if (processInstanceState != ExecutionStatus.READY_STOP
                        && processInstanceState != ExecutionStatus.READY_PAUSE) {
                    // failure task set invalid
                    taskInstance.setFlag(Flag.NO);
                    updateTaskInstance(taskInstance);
                    // crate new task instance
                    if (taskInstance.getState() != ExecutionStatus.NEED_FAULT_TOLERANCE) {
                        taskInstance.setRetryTimes(taskInstance.getRetryTimes() + 1);
                    }
                    taskInstance.setSubmitTime(null);
                    taskInstance.setLogPath(null);
                    taskInstance.setExecutePath(null);
                    taskInstance.setStartTime(null);
                    taskInstance.setEndTime(null);
                    taskInstance.setFlag(Flag.YES);
                    taskInstance.setHost(null);
                    taskInstance.setId(0);
                }
            }
        }
        taskInstance.setExecutorId(processInstance.getExecutorId());
        taskInstance.setProcessInstancePriority(processInstance.getProcessInstancePriority());
        taskInstance.setState(getSubmitTaskState(taskInstance, processInstanceState));
        if (taskInstance.getFirstSubmitTime() == null) {
            // 延时时间是根据FirstSubmitTime计算出来的，SubmitTime赋值给FirstSubmitTime，但是SubmitTime这时也为null
            taskInstance.setFirstSubmitTime(taskInstance.getSubmitTime());
        }
        boolean saveResult = saveTaskInstance(taskInstance);
        if (!saveResult) {
            return null;
        }
        return taskInstance;
    }
```

### Version

2.0.4

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by caishunfeng** [comment]
associate issue: #8685

**Comment by Tianqi-Dotes** [comment]
I have checked this issue, It is not work in version 2.0.5 but in dev is fine.
The problem is task info doesn't contain the firstSubmitTime value and is null (I guess the reason is the task don't have a submitTime, so the code cant assign firstSubmitTime  according to submitTime), so that the code can't calculate the remainTime to decide the delay
![image](https://user-images.githubusercontent.com/36755957/159294814-8956e4f4-7387-4dc9-ae77-c3caae89eb26.png)

![9f4704a62d4feb64e4304ad32b9ded8](https://user-images.githubusercontent.com/36755957/159294189-8db33306-4452-404a-900e-f44ce63a096f.png)

**Comment by DHBin** [comment]
你好,@caishunfeng,在#9085的pr中，你在submitTaskInstanceToDB方法中设置了SubmitTime，但我不觉得这是一个好的解决方法，因为先保存了SubmitTime，然后在分发任务，把任务放进了队列中，当取出任务处理后再更新了一次SubmitTime，导致这值不一致。我觉得是否考虑在TaskExecutionContext中添加多一个SubmitTime字段，解决这个不一致的问题。另一个问题是，我看了下dev分支对host的处理，我有一个疑问，为什么`TaskPriorityQueueConsumer#dispatch`中不更新host了，而是在ack中更新，期待你的解答

---- 翻译

hi, @caishunfeng, in the pr of #9085, you set the SubmitTime in the submitTaskInstanceToDB method, but I don't think this is a good solution, because the SubmitTime is saved first, and then the task is distributed and the task is put into the queue , the SubmitTime is updated once after the task is taken out and processed, resulting in inconsistent values. I think whether to consider adding one more SubmitTime field to TaskExecutionContext to solve this inconsistency problem. Another question is, I looked at the handling of the host by the dev branch. I have a question, why the host is not updated in `TaskPriorityQueueConsumer#dispatch`, but is updated in ack, looking forward to your answer

**Comment by caishunfeng** [comment]
> 你好,@caishunfeng,在#9085的pr中，你在submitTaskInstanceToDB方法中设置了SubmitTime，但我不觉得这是一个好的解决方法，因为先保存了SubmitTime，然后在分发任务，把任务放进了队列中，当取出任务处理后再更新了一次SubmitTime，导致这值不一致。我觉得是否考虑在TaskExecutionContext中添加多一个SubmitTime字段，解决这个不一致的问题。另一个问题是，我看了下dev分支对host的处理，我有一个疑问，为什么`TaskPriorityQueueConsumer#dispatch`中不更新host了，而是在ack中更新，期待你的解答
> 
> ---- 翻译
> 
> hi, @caishunfeng, in the pr of #9085, you set the SubmitTime in the submitTaskInstanceToDB method, but I don't think this is a good solution, because the SubmitTime is saved first, and then the task is distributed and the task is put into the queue , the SubmitTime is updated once after the task is taken out and processed, resulting in inconsistent values. I think whether to consider adding one more SubmitTime field to TaskExecutionContext to solve this inconsistency problem. Another question is, I looked at the handling of the host by the dev branch. I have a question, why the host is not updated in `TaskPriorityQueueConsumer#dispatch`, but is updated in ack, looking forward to your answer

Thanks for your reply. I agree with you about double save submitTime. In the original design, submitTime means that the time submit to DB. Maybe add the new field such as 'dispatchTime' is better.
Because of the wrong process, pr was directly merged into the 2.0.x branch and did not feedback to dev.(Normal process should change dev and cherry-pick to 2.0.x branch), so dev lost this change. But I had submited a pr #8719 which also do the same change, you can take a look, thanks.
